includes/wikia/api/DesignSystemApiController.class.php

```diff
@@ -231,26 +239,34 @@ public function getNavigation() {
 	 */
 	public function getAllElements() {
 		$params = $this->getRequestParameters();
+		$footerModel = new DesignSystemGlobalFooterModel(
+			$params[self::PARAM_PRODUCT],
+			$params[self::PARAM_ID],
+			$params[self::PARAM_LANG]
+		);
 
 		$this->setResponseData( [
-			'global-footer' => ( new DesignSystemGlobalFooterModel( $params[ 'wikiId' ], $params[ 'lang' ] ) )->getData(),
+			'global-footer' => $footerModel->getData(),
 			'global-navigation' => $this->data
 		] );
 
 		$this->addCachingHeaders();
 	}
 
 	private function getRequestParameters() {
-		$wikiId = $this->getRequiredParam( 'wikiId' );
-		$lang = $this->getRequiredParam( 'lang' );
+		// ultimately, id will be a required param, but fall back to wikiId while transitioning
+		$id = $this->getRequiredParam( self::PARAM_ID );
+		$product = $this->getRequiredParam( self::PARAM_PRODUCT );
+		$lang = $this->getRequiredParam( self::PARAM_LANG );
 
-		if ( WikiFactory::IDtoDB( $wikiId ) === false ) {
-			throw new NotFoundApiException( "Unable to find wiki with ID {$wikiId}" );
+		if ( $product === self::PRODUCT_WIKIS && WikiFactory::IDtoDB( $id ) === false ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)